torch/_dynamo/variables/ctx_manager.py

```diff
@@ -116,20 +117,32 @@ def call_function(
             return WrappedUserFunctionVariable(args[0], self)
 
 
-class GenericContextWrappingVariable(ContextWrappingVariable):
-    def __init__(self, target_values, initial_values=None, *, cm_obj=None, **kwargs):
+class GenericContextWrappingVariable(
+    ContextWrappingVariable, UserDefinedObjectVariable
+):
+    # Some methods in ContextWrappingVariable assumes the arguments are
+    # python contants. Which might not always be the case here.
+    def __init__(self, cm_obj, **kwargs):
         assert cm_obj is not None
         super().__init__(
-            target_values=target_values, initial_values=initial_values, **kwargs
+            target_values=None,
+            initial_values=None,
+            value=cm_obj,
+            value_type=cm_obj.__class__,
+            **kwargs,
         )
         self.cm_obj = cm_obj
 
+    def reconstruct(self, codegen):
+        # throw NotImplementedError
+        UserDefinedObjectVariable.reconstruct(self, codegen)
```

Comment:
It's a bit confused to redirect to UDO's reconstruct method, you can just throw exception explicitly.

Reply:
Maybe I should file a separate issue for this. But I found it counterintuitive that UDO's reconstruct method returns NotImplemetentedError, but we ARE able to reconstruct it via side effects

Reply:
@yanboliang, I'm redirecting instead of raising the exception here because if someone decides to update the UDO code in the future, we don't have to change anything in the subclass.